File: 고득점 kit/완주하지 못한 선수.swift
import Foundation

func solution(participant: [String], completion: [String]) -> String{
    var answer: String = ""
    var dict: [String:Int] = [:]

    for p in participant{
        // 참가중 있다면
        // 새로운 표현 습득
        if let _ = dict[p] {
            dict[p] = dict[p]! + 1
        }else{ // 없는 경우
            dict[p] = 1
        }
    }
    
    // 완주자들 체크
    for c in completion{
        dict[c] = dict[c]! - 1
    }
    
    let arr = dict.sorted(by: {$0.value < $1.value })
    answer = arr.last!.key
    return answer
}

File: 고득점 kit/해시/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string, int> hash;
    for (auto p : participant) {
        if (hash.end() == hash.find(p))
            hash.insert(make_pair(p, 1));
        else
            hash[p]++;
    }
    
    for (auto c : completion) {
        hash[c]--;
    }
    
    for (auto p : participant) {
        if (hash[p] > 0) {
            answer = p;
            break;
        }
    }
    return answer;
}